.buildkite/common.py

```diff
@@ -307,13 +309,18 @@ def build_group(self, *args, **kwargs):
         combined = overlay_dict(self.per_instance, kwargs)
         return self.add_step(group(*args, **combined), decorate=decorate)
 
-    def build_group_per_arch(self, *args, **kwargs):
+    def build_group_per_arch(self, label, *args, **kwargs):
         """
         Build a group, parametrizing over the architectures only.
         """
         decorate = kwargs.pop("decorate", True)
         combined = overlay_dict(self.per_arch, kwargs)
-        return self.add_step(group(*args, **combined), decorate=decorate)
+        key = kwargs.pop("key", None)
```

Comment:
may want to move it one line above so the `combined` dict does not have they `key` key :)

Reply:
Ah, yes, good catch. I've also added a comment about which kwargs are popped before the call to `group`